packages/markerikson-stack-client-prototype/src/pages/index.tsx

```diff
@@ -1,12 +1,19 @@
+import React, { Suspense } from "react";
 import type { NextPage } from "next";
 import styles from "../styles/Home.module.css";
 
 import { SessionContext } from "../contexts/SessionContext";
 import { useContext } from "react";
 
+import Loader from "../components/Loader";
 import { SourcesList } from "../features/sources/SourcesList";
 import { SourceContent } from "../features/sources/SourceContent";
 
+import { store } from "../app/store";
+import { setStore } from "../features/sources/sourcesCache";
+
+setStore(store);
```

Comment:
Potentially, yeah, although this is technically a pattern we do recommend for cases where non-React-related modules need access to the store:

https://redux.js.org/faq/code-structure#how-can-i-use-the-redux-store-in-non-component-files

In theory, having it get injected from a root file should mean no import order issues.

Reply:
I think it's good to avoid globals as much as possible.

Even injecting it from the (application) "root" could lead to awkward cases when writing tests for a component, for example. If we don't require/render the root, but the component uses the hook– then it's going to run into an undefined value error.